File: algorithms/maths/is_prime.py
import math

def is_prime(n) :
	if n == 1:
		return False
	if n == 2:
		return True
	if n%2 == 0:
		return False
	
	upto = int(math.sqrt(n))
	for i in range(3, upto+1, 2):
		if n%i == 0 :
			return False

	return True

#Tests
assert(is_prime(3) == True)
assert(is_prime(10) == False)
assert(is_prime(1297561) == True)

File: isPrime.js
function isPrime(n) {
   if((n > 2 && n % 2 == 0) || n == 1) {
            return false;
        }

        for (int i = 3; i <= (int)Math.sqrt(n); i += 2) {

            if (n % i == 0) {
                return false;
            }
        }

        return true;
}
